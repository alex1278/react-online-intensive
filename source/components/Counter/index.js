// Core
import React from 'react';
import { number } from 'prop-types';

// Instruments
import Styles from './styles.m.css';

// Components
// import { withProfile } from 'components/HOC/withProfile';

const Counter = ({ count }) => (
    <section className = { Styles.counter}>Posts count: {count}</section>
);

Counter.propTypes = {
    count: number.isRequired,
};

export default Counter;