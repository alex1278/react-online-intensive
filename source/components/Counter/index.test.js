// Core
import React from 'react';
import Counter from './';
import Renderer from 'react-test-renderer';

const renderTree = Renderer.create(<Counter count = { 3 }/>).toJSON();

describe('instruments', () => {
	test('Counter component sould correspond to its snapshot counterpart', () => {
		expect(renderTree).toMatchSnapshot();
	});
});