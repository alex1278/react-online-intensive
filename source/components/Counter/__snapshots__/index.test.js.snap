// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instruments Counter component sould correspond to its snapshot counterpart 1`] = `
<section
  className="counter"
>
  Posts count: 
  3
</section>
`;
